src/goto-symex/symex_assign.cpp

```diff
@@ -36,6 +36,14 @@ void goto_symext::symex_assign_rec(
   code_assignt deref_code=code;
 
   clean_expr(deref_code.lhs(), state, true);
+  // make the structure of the lhs as simple as possible to avoid,
```

Comment:
Hmm, just a brief question about your comment: what do you refer to as "it" when saying "it would get called?"